component/distributed_cache/distributed_cache.go

```diff
@@ -93,11 +132,108 @@ func (distributedCache *DistributedCache) SetNextComponent(nextComponent interna
 
+// setupCacheStructure checks and creates necessary cache directories and metadata.
+// It's doing 4 rest api calls, 3 for directory and 1 for creator file.+1 call to check the creator file
+func (dc *DistributedCache) setupCacheStructure(cacheDir string) error {
+	_, err := dc.azstroage.GetAttr(internal.GetAttrOptions{Name: cacheDir + "/Objects"})
+	if err != nil {
+		if os.IsNotExist(err) || err == syscall.ENOENT {
+			directories := []string{cacheDir, cacheDir + "/Nodes", cacheDir + "/Objects"}
+			for _, dir := range directories {
+				if err := dc.azstroage.CreateDir(internal.CreateDirOptions{Name: dir, ForceDirCreationDisabled: true}); err != nil {
+
+					if !bloberror.HasCode(err, bloberror.BlobAlreadyExists) {
+						return logAndReturnError(fmt.Sprintf("DistributedCache::Start error [failed to create directory %s: %v]", dir, err))
+					}
+				}
+			}
+
+		} else {
+			return logAndReturnError(fmt.Sprintf("DistributedCache::Start error [failed to read creator file: %v]", err))
+		}
+	}
+
+	ipaddr, err := getVmIp()
+	if err != nil {
+		return logAndReturnError(fmt.Sprintf("DistributedCache::Start error [Failed to get VM IP : %v]", err))
+	}
+
+	uuidVal, err := common.GetUUID()
+	if err != nil {
+		return logAndReturnError(fmt.Sprintf("DistributedCache::Start error [Failed to retrieve UUID, error: %v]", err))
+	}
+	rvList := make([]dcache.RawVolume, len(dc.cachePath))
+	for index, path := range dc.cachePath {
+		fsid, err := getBlockDeviceUUId(path)
+		if err != nil {
+			return logAndReturnError(fmt.Sprintf("DistributedCache::Start error [failed to get raw volume UUID: %v]", err))
+		}
+		// TODO{Akku} : More than 1 same fsid for rv, must fail distributed cache startup
+		totalSpace, err := common.GetTotalSpace(path)
+		if err != nil {
+			return logAndReturnError(fmt.Sprintf("DistributedCache::Start error [failed to evaluate local cache Total space: %v]", err))
+		}
+
+		rvList[index] = dcache.RawVolume{
+			NodeId:         uuidVal,
+			IPAddress:      ipaddr,
+			FSID:           fsid,
+			FDID:           "0",
+			TotalSpace:     totalSpace,
+			LocalCachePath: path,
+		}
+	}
+	err = dc.clusterManager.Start(&dcache.DCacheConfig{
+		CacheId:                dc.cacheID,
+		MinNodes:               dc.minNodes,
+		ChunkSize:              dc.chunkSize,
+		StripeSize:             dc.stripeSize,
+		NumReplicas:            dc.replicas,
+		MvsPerRv:               dc.mvsPerRv,
+		HeartbeatSeconds:       dc.hbDuration,
+		HeartbeatsTillNodeDown: dc.maxMissedHbs,
+		ClustermapEpoch:        dc.clustermapEpoch,
+		RebalancePercentage:    dc.rebalancePercentage,
+		SafeDeletes:            dc.safeDeletes,
+		CacheAccess:            dc.cacheAccess,
+	}, rvList)
+	if bloberror.HasCode(err, bloberror.BlobAlreadyExists) {
+		return logAndReturnError(fmt.Sprintf("DistributedCache::Start error [failed to create creator file: %v]", err))
+	} else {
+		return nil
+	}
+}
+
+func logAndReturnError(msg string) error {
+	log.Err(msg)
+	return errors.New(msg)
 }
```

Comment:
this is a fundamental function. If not already present, add it to say common/utils